### PR TITLE
Sync code with Android client implementation.

### PR DESCRIPTION
**Pull request by jessetilro**
Updated networking functionality in order to allow Android clients to
connect to the server.

### CODE REVIEW

File: src/main/java/pixelperfect/Game.java [hunk 1]
```diff
@@ -62,14 +68,33 @@ public void simpleInitApp() {
...
+   * (dis)connect.
+   */
+  public void debugNetworking() {
+    int clients = server.getConnections().size();
+    if (clients > clientCounter) {
+      System.out.println("Clients have connected to the game: "
+          + Integer.toString(clients - clientCounter) + " new clients.");
+      clientCounter = clients;
+    } else if (clients < clientCounter) {
+      System.out.println("Clients have disconnected from the game: "
```
Comment (dmalarev) [inline src/main/java/pixelperfect/Game.java]: wouldn't it be better to use a connectionListener here?

Comment (jessetilro) [inline src/main/java/pixelperfect/Game.java]: Good idea, would you like to refactor this yourself after pulling it in your branch?

Comment (dmalarev) [inline src/main/java/pixelperfect/Game.java]: Sure

**Comment by dmalarev** [comment]
i changed it to use a connection listener now.
I may have missed some checkstyle errors/

**Comment by dmalarev** [comment]
I left the old code as comments in case we need to revert

**Comment by jessetilro** [comment]
You don't need to comment out code, we can just revert the commits if necessary.

**Comment by dmalarev** [comment]
oh, right.
haven't used github in a while :D

**Comment by jessetilro** [comment]
Works like a charm, well done. I will remove the commented code and I suggest we merge so you can finish the "network" branch.

**Pull request merged**